hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/bucket/InfoBucketHandler.java

```diff
@@ -40,7 +41,56 @@ public void execute(OzoneClient client, OzoneAddress address)
         .getVolume(address.getVolumeName())
         .getBucket(address.getBucketName());
 
-    printObjectAsJson(bucket);
+    if (bucket.getSourceBucket() != null && bucket.getSourceVolume() != null) {
+      printObjectAsJson(new LinkBucket(bucket));
+    } else {
+      printObjectAsJson(bucket);
+    }
+  }
+
+  /**
+   * Class used for link buckets.
+   */
+  private static class LinkBucket {
```

Comment:
Just a thought, can we clear the metadata map from the OzoneBucket if it is a link, instead of creating a new class?

Reply:
It has other fields like quota, storage type etc., for link buckets we created with source bucket/source volume.
As I do-not want to show any other metadata taken this approach. And also it is for CLI, i don't think it is a costly operation, so we should be fine